opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/BaseInteractor.java

```diff
@@ -259,12 +263,20 @@ public void run() {
                     geometry.setCoordinates(coordinates);
                     structure.setGeometry(geometry);
                     LocationProperty properties = new LocationProperty();
-                    String structureType = event.findObs(null, false, JsonForm.STRUCTURE_TYPE).getValue().toString();
+                    String structureType = event.findObs(null, false, STRUCTURE_TYPE).getValue().toString();
                     properties.setType(structureType);
                     properties.setEffectiveStartDate(now);
                     properties.setParentId(operationalAreaId);
                     properties.setStatus(LocationProperty.PropertyStatus.PENDING_REVIEW);
                     properties.setUid(UUID.randomUUID().toString());
+                    Obs structureNameObs = event.findObs(null, false, STRUCTURE_NAME);
+                    if (structureNameObs != null && structureNameObs.getValue() != null) {
+                        properties.setName(structureNameObs.getValue().toString());
+                    }
+                    String physicalType = event.findObs(null, false, PHYSICAL_TYPE).getValue().toString();
```

Comment:
Also check for NPE physicalType observation will be null there there are app crashes when adding Mosquito and Larval sites